consensus/message.go

```diff
@@ -292,7 +295,10 @@ func (consensus *Consensus) requestBlockProposalMessageHandler(remoteMessage *no
 		if msgBody.ShortHashSize > sha256.Size {
 			return replyBuf, false, fmt.Errorf("hash size %d is greater than %d", msgBody.ShortHashSize, sha256.Size)
 		}
-		b = &block.Block{Header: fullBlock.Header}
+		b = &block.Block{
+			Header:       fullBlock.Header,
+			Transactions: []*transaction.Transaction{fullBlock.Transactions[0]},
```

Comment:
do we need to make sure that `len(fullblock.Transcations) > 0`?

Reply:
No need because it's optional